### PR TITLE
#167190397 Users can View and Edit profile 167190397

### PR DESCRIPTION
**Pull request by chiboycalix**
> - After login in, navigate to `localhost:9090/profile`
> - Click on the `Edit profile` button and a modal will be opened
> - Edit your profile by entering your new details
> - Click on save to successfully edit your profile
#### What are the relevant pivotal tracker stories?
[#167190397](https://www.pivotaltracker.com/story/show/167190397)
[e2e test link](https://drive.google.com/file/d/1GeZxJz_f_q1fwysNlZ-AiM2p4O-NWCYb/view?usp=sharing)

### CODE REVIEW

**Review by encodedBicoding — COMMENTED** [review]
Hello Nonso, Thank you for this PR, The thought process towards this is impressive. However, the image attached to this comment has a dot on it. When I click on that spot, the file input gets clicked. Was this intentional?
Screenshot 2019-09-05 at 9 21 18 PM

**Comment by chiboycalix** [comment]
@encodedBicoding that was an error from here, I have fixed it so you can test again

**Review by encodedBicoding — APPROVED** [review]
LGTM

**Review by chuxmykel — APPROVED** [review]
This is nice Nonso. Great job. Looks good to me. 
Screenshot 2019-09-05 at 23 17 36

File: src/views/ProfilePage/Profile.js [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Loader from 'react-loader-spinner';
+import Header from '@Components/Header/index';
+import Button from '@Components/Button/index';
+import Footer from '@Components/Footer/index';
+import Dialog from '@Components/Dialog/index';
+import Input from '@Components/Input/index';
+import TextArea from '@App/components/TextArea/TextArea';
+import ArticleCard from '@Components/ArticleCard/index';
```
Comment (topseySuave) [inline src/views/ProfilePage/Profile.js]: Please when you are import do not do this `import Dialog from '@Components/Dialog/index';` or 
`import TextArea from '@App/components/TextArea/TextArea';` I mentioned this before.

You import like this `import TextArea from '@App/components/TextArea';` the index file is automatically called which calls the `TextArea.js` file.

Please I don't want to make mention of this again to the entire team.

File: src/views/ProfilePage/Profile.js [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+              <div className="profile-image">
+                <img src={image} alt="" className="profile-image" />
+              </div>
+            </div>
+          </div>
+          <div />
+          <div className="profile-links">
+            <ul>
+              <li className="active">
+                <a href="/">Profile</a>
```
Comment (topseySuave) [inline src/views/ProfilePage/Profile.js]: Why isn't the React `Link` used?

**Review by topseySuave — CHANGES_REQUESTED** [review]

File: src/views/ProfilePage/Profile.js (hunk 1 above)
Comment (chiboycalix) [reply src/views/ProfilePage/Profile.js]: noted, I will make the corrections now

File: src/views/ProfilePage/Profile.js (hunk 2 above)
Comment (chiboycalix) [reply src/views/ProfilePage/Profile.js]: It was an oversight, I'll change it now.

**Review by obayomi96 — APPROVED** [review]
Good job on this task  @chiboycalix this is a job well done. Can you kindly improve the tests as well?

File: src/views/ProfilePage/Profile.js (hunk 1 above)
Comment (topseySuave) [reply src/views/ProfilePage/Profile.js]: Did you make the necessary changes? @chiboycalix

File: src/components/TextArea/TextArea.scss [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  box-shadow: 0 0 5px rgb(238, 81, 81);
+  border: 1px solid rgb(238, 81, 81);
+}
+
+.error-div {
+  margin-bottom: 5px;
+  font-family: 'Inconsolata', monospace;
+  color: red;
+  max-width: 316px;
+}
```
Comment (topseySuave) [inline src/components/TextArea/TextArea.scss]: This needs a new line.

File: src/components/Dialog/Dialog.scss [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+.modal-content-profile {
+  background-color: #fefefe;
+  margin: auto;
+  padding: 20px;
+  border: 1px solid #888;
+  width: 70%;
+  height: 70%;
+}
```
Comment (topseySuave) [inline src/components/Dialog/Dialog.scss]: this needs a new line.

**Review by topseySuave — CHANGES_REQUESTED** [review]

File: src/views/ProfilePage/Profile.js (hunk 1 above)
Comment (chiboycalix) [reply src/views/ProfilePage/Profile.js]: yes I did.

**Review by topseySuave — APPROVED** [review]

**Pull request merged**